ecommerce/frontend/src/components/SignUpForm.js

```diff
@@ -131,159 +167,187 @@ export const SignUpForm = ({ onSubmit = () => { } }) => {
+                                    message: 'Wrong captcha',
+                                },
                             ]}
-                        >
+                            hasFeedback>
                             <Input />
                         </Form.Item>
                     </Col>
                     <Col span={12}>
-                        <Button>Get captcha</Button>
+                        <LoadCanvasTemplate />
                     </Col>
                 </Row>
             </Form.Item>
 
-            {/* User type selection */}
-            <Form.Item
-                name="userType"
-                rules={[
-                    {
-                        required: true,
-                        message: "Please select your user type!"
-                    },
-                ]}
-                {...tailFormItemLayout}
-            >
-                <Radio.Group>
-                    <Radio value="customer">Register as customer</Radio>
-                    <Radio value="vendor">Register as vendor</Radio>
-                </Radio.Group>
-            </Form.Item>
-
             {/* Checkbox for the user agreement */}
             <Form.Item
-                name="agreement"
-                valuePropName="checked"
-                rules={[
-                    {
-                        validator: (_, value) =>
-                            value ? Promise.resolve() : Promise.reject('Should accept agreement'),
-                    },
+                {...tailFormItemLayout}>
+                <Form.Item
+                    name="agreement"
+                    valuePropName="checked"
+                    rules={[
+                        {
+                            validator: (_, value) =>
+                                value ? Promise.resolve() : Promise.reject('Should accept agreement.'),
+                        },
+                    ]}>
+                    <Checkbox checked={userAgreementAccepted}>
+                        I have read the &nbsp;
+                        <Button type="dashed"  onClick={() => setUserAgreementVisible(true)}> 
+                            User Agreement
+                        </Button>   
+                    </Checkbox>
+                </Form.Item>
+                
+                
+                <Modal
+                    visible={userAgreementVisible}
+                    title="User Agreement"
+                    onOk={onModalAgreementAccept}
+                    onCancel={onModalAgreementReject}
+                    bodyStyle={{height: 500}}
+                    footer={[
+                <Button key="back" type="primary" onClick={onModalAgreementReject}>
+                    I reject
+                </Button>,
+                <Button key="submit" type="primary" onClick={onModalAgreementAccept}>
+                    I accept
+                </Button>,
```

Comment:
not sure if you need the callbacks both on onOK and on the button itself. It might make the callback be called twice.

Reply:
I have checked for double callback case. It only calls them once. So I am not going to change anything here.